include/swift/AST/Decl.h

```diff
@@ -5899,9 +5907,15 @@ class AbstractFunctionDecl : public GenericContext, public ValueDecl {
     Bits.AbstractFunctionDecl.IAMStatus = newValue.getRawValue();
   }
 
+  /// Retrieve the location of the 'async' keyword, if present.
+  SourceLoc getAsyncLoc() const { return AsyncLoc; }
+
   /// Retrieve the location of the 'throws' keyword, if present.
   SourceLoc getThrowsLoc() const { return ThrowsLoc; }
 
+  /// Returns true if the function is async.
+  bool hasAsync() const { return Bits.AbstractFunctionDecl.Async; }
```

Comment:
Some of the features we're considering distinguish between functions that behave like `async` functions for their callers and those that expose an `async` context within the implementation.  (This is useful for e.g. UI event handlers.) This should probably be clear about which it means.